### ISSUE TITLE
Unable to connect to tiller on AKS from within cluster

### ISSUE THREAD

**Issue by tirumaraiselvan** [kubernetes/helm#4254]
I make Gitkube (https://github.com/hasura/gitkube). Gitkube is a tool that builds apps on the cluster and deploys it. It supports installing helm charts. 

I am seeing an issue on AKS which does not happen on GKE.

On GKE, everything is fine. When I do `helm version` from inside the gitkubed pod, I get expected results

```
Client: &version.Version{SemVer:"v2.9.1", GitCommit:"20adb27c7c5868466912eebdf6664e7390ebe710", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.9.1", GitCommit:"20adb27c7c5868466912eebdf6664e7390ebe710", GitTreeState:"clean"}
```

On AKS though the story is different. `helm version` does not resolve tiller. Is it because of the tiller that comes pre-installed on AKS clusters? Is it because of some other AKS specific issue? 

Output of `helm version`:
```
$ helm version
Client: &version.Version{SemVer:"v2.9.1", GitCommit:"20adb27c7c5868466912eebdf6664e7390ebe710", GitTreeState:"clean"}

Stuck finding server version
```

Output of `kubectl version`:

```
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"8", GitVersion:"v1.8.7", GitCommit:"b30876a5539f09684ff9fde266fda10b37738c9c", GitTreeState:"clean", BuildDate:"2018-01-16T21:59:57Z", GoVersion:"go1.8.3", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"9", GitVersion:"v1.9.6", GitCommit:"9f8ebd171479bec0ada837d7ee641dec2f8c6dd1", GitTreeState:"clean", BuildDate:"2018-03-21T15:13:31Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}
```

Cloud Provider/Platform (AKS, GKE, Minikube etc.): 

AKS v1.9.6 ( tiller comes pre-baked)

**Comment by bacongobbler** [comment]
I just booted a fresh AKS cluster and had no issues.

```
><> az aks get-credentials -n helmtest -g helmtest
Merged "helmtest" as current context in /home/bacongobbler/.kube/config
><> kk get po
NAME                                    READY     STATUS    RESTARTS   AGE
heapster-55f855b47-ns4dn                2/2       Running   0          5m
kube-dns-v20-7c556f89c5-2wllt           3/3       Running   0          6m
kube-dns-v20-7c556f89c5-jl26r           3/3       Running   0          6m
kube-proxy-6q5jq                        1/1       Running   0          6m
kube-proxy-j8vzp                        1/1       Running   0          6m
kube-proxy-nd9dz                        1/1       Running   0          6m
kube-svc-redirect-bjt8k                 1/1       Running   0          6m
kube-svc-redirect-g45pp                 1/1       Running   0          6m
kube-svc-redirect-qcptv                 1/1       Running   0          6m
kubernetes-dashboard-546f987686-hqb65   1/1       Running   0          6m
tunnelfront-78698f557d-snglm            1/1       Running   0          6m
><> helm init
$HELM_HOME has been configured at /home/bacongobbler/.helm.

Tiller (the Helm server-side component) has been installed into your Kubernetes Cluster.

Please note: by default, Tiller is deployed with an insecure 'allow unauthenticated users' policy.
For more information on securing your installation see: https://docs.helm.sh/using_helm/#securing-your-helm-installation
Happy Helming!
><> kk get po
NAME                                    READY     STATUS    RESTARTS   AGE
heapster-55f855b47-ns4dn                2/2       Running   0          5m
kube-dns-v20-7c556f89c5-2wllt           3/3       Running   0          6m
kube-dns-v20-7c556f89c5-jl26r           3/3       Running   0          6m
kube-proxy-6q5jq                        1/1       Running   0          6m
kube-proxy-j8vzp                        1/1       Running   0          6m
kube-proxy-nd9dz                        1/1       Running   0          6m
kube-svc-redirect-bjt8k                 1/1       Running   0          6m
kube-svc-redirect-g45pp                 1/1       Running   0          6m
kube-svc-redirect-qcptv                 1/1       Running   0          6m
kubernetes-dashboard-546f987686-hqb65   1/1       Running   0          6m
tiller-deploy-7ccf99cd64-64lzp          0/1       Running   0          6s
tunnelfront-78698f557d-snglm            1/1       Running   0          6m
><> helm version
Client: &version.Version{SemVer:"v2.9.1", GitCommit:"20adb27c7c5868466912eebdf6664e7390ebe710", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.9.1", GitCommit:"20adb27c7c5868466912eebdf6664e7390ebe710", GitTreeState:"clean"}
```

I'm not sure what would be the isue, but perhaps it'd be best to open a ticket at https://github.com/Azure/AKS as they would be more likely able to help you with any connection issues.

For reference, all Helm does is a `kubectl port-forward` behind the scenes to initiate a connection, and it has been confirmed to work on pretty much any (working) Kubernetes cluster at this point :)

**Comment by bacongobbler** [comment]
I'm going to close this as there doesn't appear to be anything actionable on our end (the issue looks to be on AKS's end), but let's re-open this if something else comes up. Thanks!

**Comment by tirumaraiselvan** [comment]
@bacongobbler As mentioned in the report, the helm client is running inside a pod in the cluster. It is a different setup.

**Comment by tirumaraiselvan** [comment]
I should have mentioned, when I try helm upgrade on a chart I get:

```
Error: forwarding ports: error upgrading connection: Upgrade request required
```

**Comment by bacongobbler** [comment]
RIght, though the difference between running the helm client locally and running inside a pod aren't going to make much difference. It's going to reach out to the k8s API to attain a port-forward connection to tiller, then use that.

What's the output of `helm version --debug` from the pod?

**Comment by tirumaraiselvan** [comment]
`helm version --debug` printed the client version and then got stuck again with no new information.

The `port-forward` command was quite revealing. It threw the exact same error again:

`
Error: forwarding ports: error upgrading connection: Upgrade request required
`

Made me think if this is a firewall issue where port 44134 is not allowed. So I went and opened that port on all the nodes but to no avail. I guess the host network is different from the pod network and the firewall rules didn't get applied to the pods.

Anyway, this is definitely an AKS issue because this works allright it GKE. I have raised an issue here: https://github.com/Azure/AKS/issues/459

Thanks for your help!
